node/src/components/consensus/utils.rs

```diff
@@ -129,12 +129,12 @@ where
 pub(crate) fn check_sufficient_finality_signatures(
     trusted_validator_weights: &BTreeMap<PublicKey, U512>,
     finality_threshold_fraction: Ratio<u64>,
-    block_signatures: &BlockSignatures,
+    block_signatures: Option<&BlockSignatures>,
```

Comment:
Is that really simpler than not calling the function at all if we don't have any signatures?

Reply:
I agree with Andreas - I think it's a bit weird to make the function check if we even supplied any signatures at all. Obviously they are not sufficient then, and the caller can check for that case.

Reply:
It is easier to bubble up the `FinalitySignatureError::InsufficientWeightForFinality` than to create it manually on the call site, where we'd need to, for example, calculate the `total_weight`.

Reply:
Alternatively we could just pass in empty `BlockSignatures` instead of `None`?